tracing/trace_state.hh

```diff
@@ -390,9 +390,9 @@ private:
      */
     void add_prepared_statement(prepared_checked_weak_ptr& prepared);
 
-    void set_username(const std::optional<auth::authenticated_user>& user) {
+    void set_user(lw_shared_ptr<auth::authenticated_user> user) {
         if (user) {
-            _records->session_rec.username = format("{}", *user);
+            _records->session_rec.user = std::move(user);
```

Comment:
There's something I just realized I didn't understand about this series.
I thought that the "hot path" of tracing was when tracing is NOT enabled - we don't want to do a lot of allocations, copies, etc., when the result of all of that isn't even used. But if this is what we want to optimize, we can simply do nothing if tracing is disabled. Why don't we do that optimization?
Or is the purpose of this series to improve "fast slow-query tracing" - where tracing is always enabled but usually not saved anywhere?

Reply:
> But if this is what we want to optimize, we can simply do nothing if tracing is disabled. Why don't we do that optimization?

When tracing is disabled, all tracing calls do nothing. So it's already optimized.

The `hot path` I am referring to is the optimization of the execution of the most popular code path that I assume is - an execution of a single prepared statement. In that "hot path" I am trying to optimize the case when we enable _lightway slow-query tracing_ to be as invisible as it can be to the outer performance.

Reply:
I see, then as I guessed you are trying to optimize the "lightweight slow-query tracing", where we always start tracing for each and every query but only on the coordinator node (hence the lightweight) and in almost every case we throw away the saved information. Maybe you should mention this in the commit messae.